### ISSUE TITLE
about loss weight

### ISSUE THREAD

**Issue by konioy** [ultralytics/yolov5#8188]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Why is it set like this？
    hyp['box'] *= 3 / nl  # scale to layers
    hyp['cls'] *= nc / 80 * 3 / nl  # scale to classes and layers
    hyp['obj'] *= (imgsz / 640) ** 2 * 3 / nl  # scale to image size and layers
default  hyp['box'] : hyp['cls'] :  hyp['obj'] = 1:1:1
if my data nc is 3, hyp['box'] : hyp['cls'] :  hyp['obj'] = 1: 3/80 :1????

**Comment by Zephyr69** [comment]
If your dataset has 3 classes, the cls loss is going to start overfitting way earlier than with a dataset of 80 classes (with all else being equal). This usually (cuz u may also need to consider the values of the other two losses) won't end well in terms of model metrics.

**Comment by konioy** [comment]
Thanks for your reply, I understand
Why use 80 as the benchmark

**Comment by konioy** [comment]
Thank you for your reply, why is it based on 80？

**Comment by glenn-jocher** [comment]
COCO has 80 classes

**Comment by konioy** [comment]
Why use coco data as a benchmark. In other words, why should we make proportional changes based on three detection layers, 80 classes, and input size of 640? Personally, I think that there is a unique value for each data, and a simple proportional change should not be performed.

**Comment by glenn-jocher** [comment]
COCO is the world’s most common reference benchmark dataset for object detection.

**Comment by konioy** [comment]
why should we make proportional changes based on 3 detection layers, 80 classes, and input size of 640? Personally, I think that there is a unique value for each data, and a simple proportional change should not be performed.

**Comment by glenn-jocher** [comment]
@konioy you don't understand, the code is in place to customize to your specific dataset. That's the entire purpose of this code.

**Comment by konioy** [comment]
Specifically, if my private data has 3 categories, 2 detect layers, and input size of 736, how do you recommend setting the loss weight?

**Comment by glenn-jocher** [comment]
@konioy the lines above do this automatically. No code customization is required on your part to train custom data. See Train Custom Data tutorial to get started.

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️  RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Roboflow for Datasets, Labeling, and Active Learning](https://docs.ultralytics.com/yolov5/tutorials/roboflow_datasets_integration)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TFLite, ONNX, CoreML, TensorRT Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [Architecture Summary](https://docs.ultralytics.com/yolov5/tutorials/architecture_description)&nbsp; ⭐ NEW

Good luck 🍀 and let us know if you have any other questions!

**Comment by konioy** [comment]
Thank you!

**Comment by glenn-jocher** [comment]
@konioy you're welcome! If you have any more questions, feel free to ask. Good luck with your custom training! 🍀
